### PR TITLE
Account info component

### PR DESCRIPTION
**Pull request by abericExecom**

### CODE REVIEW

File: client-app/src/app/app.module.ts [hunk 1]
```diff
@@ -53,6 +53,8 @@ import { ModalComponent } from './shared/components/modal/modal.component';
 import { OrderConfirmationComponent } from './pages/my-order/order-confirmation/order-confirmation.component';
 import { TranslateModule, TranslateLoader } from '@ngx-translate/core';
 import { TranslateHttpLoader } from '@ngx-translate/http-loader';
+import { AccountInfoComponent } from './pages/my-account/account-info/account-info.component';
+import { customerReducer } from './store/reducers/customer.reducer';
```
Comment (mkovacevicExecom) [inline client-app/src/app/app.module.ts:57]: use @store

File: client-app/src/app/app.module.ts [hunk 2]
```diff
@@ -91,7 +93,7 @@ const COMPONENTS = [
 const SERVICES = [UserService, RestaurantService, FoodService];
 
 @NgModule({
-  declarations: [...COMPONENTS, DropdownDirective, ClickOutsideDirective, CurrencyFormatPipe],
+  declarations: [...COMPONENTS, DropdownDirective, ClickOutsideDirective, CurrencyFormatPipe, AccountInfoComponent],
```
Comment (mkovacevicExecom) [inline client-app/src/app/app.module.ts:96]: maybe you should move AccountInfoComponent in COMPONENTS list

File: client-app/src/app/core/models/meal.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+export interface IMeal {
+  id: number;
+  name: string;
+  type: string;
+  price: number;
+  description: string;
+  note: string;
+  condiments: string[];
```
Comment (mkovacevicExecom) [inline client-app/src/app/core/models/meal.ts:8]: maybe Array<string> for type

File: client-app/src/app/pages/my-account/account-info/account-info.component.scss [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  @media only screen and (max-width: $bp-medium) {
+    font-size: 1.4rem;
+    width: 100%;
+  }
+}
+
+.user {
+  &-image {
+    height: 12.2rem;
+    width: 12.2rem;
```
Comment (mkovacevicExecom) [inline client-app/src/app/pages/my-account/account-info/account-info.component.scss:15]: min-width?

File: client-app/src/app/pages/my-account/account-info/account-info.component.scss [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    border: 0.1rem solid $color-grey-light-2;
+    border-radius: 50%;
+
+    @media only screen and (max-width: $bp-medium) {
+      height: 8rem;
+      width: 8rem;
```
Comment (mkovacevicExecom) [inline client-app/src/app/pages/my-account/account-info/account-info.component.scss:21]: same here

File: client-app/src/app/pages/my-account/account-info/account-info.component.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-account-info',
+  templateUrl: './account-info.component.html',
+  styleUrls: ['./account-info.component.scss'],
+})
+export class AccountInfoComponent {
+  public customer$: Observable<ICustomer> = this.store.select(getCustomer);
+  constructor(private store: Store<IStore>) {}
```
Comment (mkovacevicExecom) [inline client-app/src/app/pages/my-account/account-info/account-info.component.ts:16]: /n before constructor

File: client-app/src/app/pages/my-account/my-account.component.scss [hunk 7]
```diff
@@ -1,59 +1,5 @@
 @import 'styles';
```
Comment (mkovacevicExecom) [inline client-app/src/app/pages/my-account/my-account.component.scss:1]: Is this import necessary?

File: client-app/src/app/store/reducers/customer.reducer.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+    ...state,
+    id: action.payload.id,
+    name: action.payload.name,
+    balance: action.payload.balance,
+    defaultLocation: action.payload.defaultLocation,
+    googleUserId: action.payload.googleUserId,
+    // @TODO remove favouriteMeals and add favouriteRestaurants when backend changes customer object
+    favouriteMeals: action.payload.favouriteMeals,
+    orders: action.payload.orders,
+    photoUrl: action.payload.photoUrl,
```
Comment (mkovacevicExecom) [inline client-app/src/app/store/reducers/customer.reducer.ts:29]: you should use ...action.payload, in that case, you don't need to use  id: action.payload.id...

File: client-app/src/app/pages/my-account/account-info/account-info.component.scss (hunk 4 above)
Comment (abericExecom) [reply client-app/src/app/pages/my-account/account-info/account-info.component.scss:15]: Its an Image, Beki said you don't min-width, the content is not changing.

File: client-app/src/app/pages/my-account/account-info/account-info.component.scss (hunk 5 above)
Comment (abericExecom) [reply client-app/src/app/pages/my-account/account-info/account-info.component.scss:21]: Same as above

File: client-app/src/app/store/reducers/customer.reducer.ts [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+  }))
+);
```
Comment (mkovacevicExecom) [inline client-app/src/app/store/reducers/customer.reducer.ts:31]: maybe you should add a user remove action, we will need it when logging out?

File: client-app/src/app/pages/my-account/my-account.component.scss (hunk 7 above)
Comment (abericExecom) [reply client-app/src/app/pages/my-account/my-account.component.scss:1]: Yes, I'm using breakpoints

File: client-app/src/app/store/reducers/customer.reducer.ts (hunk 8 above)
Comment (abericExecom) [reply client-app/src/app/store/reducers/customer.reducer.ts:29]: Uuuu, i like that :D

**Review by MiljkovicVeljko — APPROVED** [review]

**Review by mkovacevicExecom — APPROVED** [review]

File: client-app/src/app/core/models/order.ts [hunk 10]
```diff
@@ -0,0 +1,8 @@
+import { IMeal } from '../models/meal';
```
Comment (aberonja) [inline client-app/src/app/core/models/order.ts:1]: /n

File: client-app/src/app/pages/home/header/header.component.ts [hunk 11]
```diff
@@ -48,6 +52,7 @@ export class HeaderComponent implements OnInit, OnDestroy {
         this.userService.fetchCustomerData(claims.email, claims.picture).subscribe(
```
Comment (aberonja) [inline client-app/src/app/pages/home/header/header.component.ts:52]: Shouldn't this be dispatch as well?

File: client-app/src/app/pages/my-account/account-info/account-info.component.html [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+  </h1>
+  <span class="user-name mt-4">{{ (customer$ | async).name }}</span>
+  <span class="user-email">{{ (customer$ | async).googleUserId }}</span>
+  <span class="user-name mt-4">{{ 'ACCOUNT.DEFAULT_LOCATION' | translate }}</span>
+  <div class="location d-flex flex-column mb-5">
+    <app-location-selector class="location-selector mt-2"></app-location-selector>
+  </div>
+  <span class="user-name mt-5">{{ 'ACCOUNT.CREDIT_BALANCE' | translate }}</span>
+  <div class="credit-balance mx-auto mt-2">
+    <span>{{ (customer$ | async).balance | currencyFormat }}</span>
```
Comment (aberonja) [inline client-app/src/app/pages/my-account/account-info/account-info.component.html:14]: You can have one async on the top. Do some research, you'll find it quickly. :)

File: client-app/src/app/shared/mock/user.mock.ts [hunk 13]
```diff
@@ -1,7 +1,16 @@
-export const USER_MOCK = {
-  id: 0,
+import { ORDERS } from '../mock/oders.mock';
+import { ICustomer } from '@core/models/customer';
```
Comment (aberonja) [inline client-app/src/app/shared/mock/user.mock.ts:2]: /n

**Review by aberonja — CHANGES_REQUESTED** [review]

File: client-app/src/app/pages/home/header/header.component.ts (hunk 11 above)
Comment (aberonja) [reply client-app/src/app/pages/home/header/header.component.ts:52]: Leave a @todo so you can implement dispatch once you add effects.

**Review by aberonja — APPROVED** [review]

**Pull request merged**